public/js/icinga/behavior/navigation.js

```diff
@@ -282,63 +314,96 @@
     };
 
     /**
-     * Show the fly-out menu
+     * Captures the mouse enter events to the navigation item and show the flyout.
      *
      * @param e
      */
-    Navigation.prototype.showFlyoutMenu = function(e) {
-        var $layout = $('#layout');
-
+    Navigation.prototype.onMouseEnter = function(e) {
+        const $layout = $('#layout');
+        const _this = e.data.self;
         if ($layout.hasClass('minimal-layout')) {
             return;
         }
 
-        var $target = $(this);
-        var $flyout = $target.find('.nav-level-2');
+        const $target = $(this);
+
+        if (! _this.coordinates.includes(undefined) && d3.polygonContains(_this.coordinates, [e.clientX, e.clientY])) {
+            return;
+        }
 
-        if (! $flyout.length) {
+        if (! $target[0].matches(':has(.nav-level-2)')) {
             $layout.removeClass('menu-hovered');
             $target.siblings().not($target).removeClass('hover');
             return;
         }
 
-        var delay = 300;
+        $layout.addClass('menu-hovered');
+        _this.coordinates[0] = [e.clientX, e.clientY];
+        _this.showFlyoutMenu($target, _this.coordinates[0]);
+    }
 
-        if ($layout.hasClass('menu-hovered')) {
-            delay = 0;
+    /**
+     * Captures the mouse move events within the navigation item
+     * and show the flyout if needed.
+     *
+     * @param e
+     */
+    Navigation.prototype.onMouseMove = function(e) {
+        const _this = e.data.self;
+        clearTimeout(_this.flyoutTimer);
+
+        const $target = $(this);
+        _this.coordinates[0] = [e.clientX, e.clientY];
+        if (! _this.coordinates.includes(undefined)) {
+            _this.triangle.attr("points", _this.coordinates.map(p => p.join(",")).join(" "));
         }
 
-        setTimeout(function() {
-            try {
-                if (! $target.is(':hover')) {
-                    return;
-                }
-            } catch(e) { /* Bypass because if IE8 */ }
+        if ($target[0].matches(':has(.nav-level-2)')) {
+            _this.flyoutTimer = setTimeout(function() {
```

Comment:
This runs out in case movement stops while still hovering the same target. Don't start the timeout in case the shown flyout is the one of the target!